viewer/packages/tools/src/Geomap/Geomap.ts

```diff
@@ -21,14 +21,20 @@ export class Geomap {
     const bound = this._viewer.models[0].getModelBoundingBox();
     this._map.position.set(-coords.x, bound.min.y, coords.y);
     this._map.updateMatrixWorld(true);
+    this.requestRedraw(10000);
+    this._viewer.on('cameraChange', () => {
+      this.requestRedraw(5000);
+    });
+  }
 
-    var intervalId = setInterval(function() {
-      viewer.requestRedraw();
+  private requestRedraw(timeOut: number) {
+    var intervalId = setInterval(() => {
+      this._viewer.requestRedraw();
     }, 100);
 
-      setTimeout(() => {
-        clearInterval(intervalId);
-      }, 20000);
+    setTimeout(() => {
+      clearInterval(intervalId);
+    }, timeOut);
```

Comment:
This will get problematic when camera is moved over a period of time. `cameraChange` will trigger many times (lets say 100) and every time you set up a new `setInterval`. Eventually there will be 100 timers requesting redraw. This is no big problem, but is not necessary so I think we should avoid it.

Instead, keep any running `intervalId` as a member variable, and make sure you call `clearInterval` before setting up any new interval.

Other than that it looks good :)